bgpd/bgp_vty.c

```diff
@@ -107,22 +107,33 @@ bgp_node_safi (struct vty *vty)
   return safi;
 }
 
-int
-bgp_parse_afi(const char *str, afi_t *afi)
+/* supports (ipv4|ipv6) */
+afi_t
+bgp_vty_afi_from_arg(const char *afi_str)
 {
-    if (!strcmp(str, "ipv4")) {
-	*afi = AFI_IP;
-	return 0;
+  afi_t afi = AFI_MAX;       /* unknown */
+  if (!strcmp(afi_str, "ipv4")) {
+    afi = AFI_IP;
     }
 #ifdef HAVE_IPV6
```

Comment:
I think removing all the HAVE_IPV6s is a good idea.  IMO this should be a standalone patch.  BTW https://github.com/Quagga/quagga/commit/205e6744f2dc2909dd494c9ce8acb82821459f1f